spec/jobs/chat_channel_delete_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Jobs::ChatChannelDelete do
+  fab!(:chat_channel) { Fabricate(:chat_channel) }
+  fab!(:user1) { Fabricate(:user) }
+  fab!(:user2) { Fabricate(:user) }
+  fab!(:user3) { Fabricate(:user) }
+  let(:users) { [user1, user2, user3] }
+
+  before do
+    messages = []
+    20.times do
+      messages << Fabricate(:chat_message, chat_channel: chat_channel, user: users.sample)
```

Comment:
The usage of `sample` here makes me nervous because the start state of every test run is going to be different. If the start state is different, we cannot rely on the failures of a previous test run to debug the next test route.

Reply:
I don't think this is a problem, `users.sample` is just used here to avoid creating tons of extra user records unnecessarily from fabricators.